__tests__/middleware.spec.ts

```diff
@@ -5,101 +5,94 @@
 import { NextRequest, NextResponse } from 'next/server';
 import { middleware } from '../middleware';
 import { verifyJwtToken } from '../lib/server/auth';
+import { getAuthTokenServer } from '@/utils/getAuthTokenServer';
+import { deleteCookie } from '@/utils/cookiesUtil';
 
 
 jest.mock('../lib/server/auth', () => ({
     verifyJwtToken: jest.fn(),
 }));
 
-const redirectSpy = jest.spyOn(NextResponse, 'redirect');
+jest.mock('@/utils/getAuthTokenServer', () => ({
+    getAuthTokenServer: jest.fn(),
+}));
 
-describe('middleware.ts', () => {
+jest.mock('@/utils/cookiesUtil', () => ({
+    deleteCookie: jest.fn()
+}))
 
-    const mockNextRequest = (url: string, cookies: Record<string, string> = {}) => {
-        return {
-            nextUrl: new URL(url, 'http://localhost:3000'),
-            cookies: {
-                get: (name: string) => (cookies[name] ? { value: cookies[name] } : undefined),
-                delete: jest.fn(),
-            },
-            headers: new Headers(),
-        } as unknown as NextRequest;
-    };
+describe('middleware.ts', () => {
+    let redirectSpy: jest.SpyInstance;
 
     beforeEach(() => {
+        redirectSpy = jest.spyOn(NextResponse, 'redirect');
         jest.clearAllMocks();
     });
 
-    it('should skip authentication for excluded paths', () => {
-        // @ts-ignore
-        const request = mockNextRequest('/api/setCookie');
-        const response = middleware(request);
-
-        expect(response).toEqual(NextResponse.next());
-    });
+    afterEach(() => {
+        redirectSpy.mockRestore();
+    })
 
-    it('should redirect to login if no token is found for protected routes', () => {
-        const request = mockNextRequest('/dmps/123');
-        middleware(request);
+    it('should redirect to login if no token is found for protected path', async () => {
+        (getAuthTokenServer as jest.Mock).mockResolvedValue(null);
+        const request = new NextRequest(new Request('http://localhost:3000/dmps/123'));
+        const response = await middleware(request);
 
-        expect(redirectSpy).toHaveBeenCalledWith(`${process.env.NEXT_PUBLIC_BASE_URL}/login`);
+        expect(redirectSpy).toHaveBeenCalledWith('http://localhost:3000/login');
```

Comment:
Just wondering if there is a reason to hard-code the localhost url here? Keeping it as an environment variable might be preferable, especially since every dev might have their environment set up differently.

Reply:
Ah, reading further into the code, looks like all of this is inside the context of a test and mocking the request/response. If this is the case then it's understandable to use a fixed url.